### ISSUE TITLE
General go deprecation

### ISSUE THREAD

**Issue by James-Pickett** [kolide/launcher#938]
A perennial topic, keeping up with go depreciations

As go has updated, we should update our code to use modern go idioms. Offhand, I can think of:
* [x] https://github.com/kolide/launcher/pull/937
* [x] [Move off `ioutil`](https://github.com/kolide/launcher/pull/946)
* [ ] `go-bindata` I purged most of this, but there's a little left in how the notary/TUF stuff is instantiated during build. Getting rid of this would smooth build somewhat. I think that's the only remaining tool outside of go itself. (well, and make)
* [ ] logging -- https://github.com/golang/go/discussions/54763

**Comment by James-Pickett** [comment]
Addresses converting `github.com/pkg/errors` to `fmt errors`: https://github.com/kolide/launcher/pull/937

**Comment by James-Pickett** [comment]
Regarding the removal of `go-bindata`, the goal here is to replace it with the native go embed? Everything else, building out the tuf directory and all that would remain the same?

**Comment by directionless** [comment]
> Regarding the removal of `go-bindata`, the goal here is to replace it with the native go embed? Everything else, building out the tuf directory and all that would remain the same?

Yes, mostly. There's a 2 phase thing, so it needs embed coupled with generate. I started this in https://github.com/kolide/launcher/compare/main...directionless:launcher:seph/no-go-bindata

I forget where I stopped. I didn't get to testing all of it. I think there was some clever thing I wanted to do about updating the embed dir, and no the tools dir, and fixing the ignores
